code/gui/serialbox.cpp

```diff
@@ -76,6 +94,76 @@ SerialBox::~SerialBox() {
     delete ui;
 }
 
+void SerialBox::slider_changed(int value, int dir) {
+    m_edit_boxes[dir]->setText(QString::number(value));
+    switch (dir) {
+        case 0:
+            m_solenoid->change_up_power(value);
```

Comment:
why don't you call a generic solenoid change power function instead?

Reply:
I do but I call it in `solenoid.cpp` instead. Should I move the switch case there?